platforms/core-runtime/base-services/src/main/java/org/gradle/internal/jvm/JpmsConfiguration.java

```diff
@@ -26,10 +26,15 @@
  * Gradle accesses those packages reflectively. On Java versions 9 to 15, the users will get
  * a warning they can do nothing about. On Java 16+, strong encapsulation of JDK internals is
  * enforced and not having the explicit permissions for reflective accesses will result in runtime exceptions.
+ *
+ * <p>
+ * For the {@code --enable-native-access} flag, users will get a warning on Java 24+ if the flag is not present.
+ * There is no enforcement of the flag at the time of writing.
+ * </p>
  */
 public class JpmsConfiguration {
```

Comment:
For readability, I think it would be nicer if...

1. All of the lists were inside the static block.  This keeps them all together vs scattered
2. All of the static lists should be assigned in the static block.  Some javadoc on these fields can describe what their purpose is
3. The getters should be together with some javadoc. You might also consider a name that looks less like getting a property and more like calculating a value.  e.g., `JpmsConfiguration.forGroovyCompiler(javaVersion)` vs `JpmsConfiguration.getGroovyArgs(javaVersion)` 

WRT the names of the fields, I think we can get away with something like:
`GRADLE_DAEMON_JPMS_ARGS_$javaVersion` and ignore the detail about native services.